.github/workflows/CD.yml

```diff
@@ -517,50 +495,57 @@ jobs:
-
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
-          aws-region: us-west-2
+          path: |
+            VERSION
+          key: release-latest-image-${{ github.run_id }}
 
       - name: Compare version with Dockerhub latest
         id: version
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
         run: |
-          TAG=`cat build/packages/VERSION`
+          TAG="${{ github.event.inputs.version }}"
           TARGET_VERSION=$TAG bash tools/workflow/docker-version-compare.sh
 
-      - name: Tag latest and upload to ecr
-        if: steps.version.outputs.any-update == 'true'
-        run: |
-          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/aws-observability
-          docker tag $IMAGE_NAME public.ecr.aws/$ECR_REPO:latest
-          docker push public.ecr.aws/$ECR_REPO:latest
-
       - name: Login Dockerhub
         uses: docker/login-action@v1
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
         with:
           username: "${{ secrets.DOCKERHUB_RELEASE_USERNAME }}"
           password: "${{ secrets.DOCKERHUB_RELEASE_TOKEN }}"
 
-      - name: Tag latest and upload to docker
-        if: steps.version.outputs.any-update == 'true'
-        run: |
-          REPO_NAME="$IMAGE_NAMESPACE/$IMAGE_NAME"
-          docker tag $IMAGE_NAME $REPO_NAME:latest
-          docker push $REPO_NAME:latest
+      - name: Login to Public Release ECR
+        id: login-ecr
+        if: ${{ steps.release-latest-image.outputs.cache-hit != 'true' && steps.version.outputs.any-update == 'true' }}
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.RELEASE_KEY_ID }}
+          password: ${{ secrets.RELEASE_SECRET }}
+        env:
+          AWS_REGION: us-east-1
+
+      - name: Pull image from integration test ECR, tag as latest and push to public release ECR and DockerHub
+        if: ${{ steps.release-latest-image.outputs.cache-hit != 'true'  && steps.version.outputs.any-update == 'true' }}
+        uses: akhilerm/tag-push-action@v2.0.0
+        with:
+          src: public.ecr.aws/${{ env.ECR_REPO }}:${{ github.event.inputs.version }}
+          dst: |
+            public.ecr.aws/${{ env.ECR_REPO }}:latest
+            $IMAGE_NAMESPACE/${{ env.IMAGE_NAME }}:latest
+
+      - name: Pull image from integration test ECR, tag with input version and push to DockerHub
+        uses: akhilerm/tag-push-action@v2.0.0
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
+        with:
+          src: public.ecr.aws/${{ env.ECR_REPO }}:${{ github.event.inputs.version }}
+          dst: $IMAGE_NAMESPACE/${{ env.IMAGE_NAME }}:${{ github.event.inputs.version }}
```

Comment:
So `$IMAGE_NAMESPACE` is fine, but `$IMAGE_NAME` is not? That's weird.

Reply:
It's not. My mistake for not checking carefully. Fixed it with the newest commit.